### PR TITLE
eth/gasprice, internal/ethapi, miner: minor feehistory fixes

### PR DESCRIPTION
**Pull request by karalabe**
This PR contains a few fixes to the feeHistory code, which was merged a bit prematurely:

- The snapshotReceipts in the miner are copied to avoid a data race.
- The API return fields are lower cased and the rewards omitted if not requested.
- Error returns are modified a bit to be consistent with Go styles and some contextual infos are added.
- Naked returns are removed since they made the code very hard to follow.
- Other nits and bits.

### CODE REVIEW

File: eth/gasprice/feehistory.go [hunk 1]
```diff
@@ -189,88 +197,87 @@ func (oracle *Oracle) resolveBlockRange(ctx context.Context, lastBlockNumber rpc
...
-	for ; blockCount > 0; blockCount-- {
-		bf := <-resultCh
+	var (
+		reward       = make([][]*big.Int, blocks)
+		baseFee      = make([]*big.Int, blocks+1)
+		gasUsedRatio = make([]float64, blocks)
+		firstMissing = blocks
+	)
+	for ; blocks > 0; blocks-- {
+		bf := <-results
```
Comment (rjl493456442) [inline eth/gasprice/feehistory.go:276]: Please also rename it to `fees`

**Review by rjl493456442 — APPROVED** [review]
nitpicks, lgtm

File: internal/ethapi/api.go [hunk 2]
```diff
@@ -80,28 +80,28 @@ func (s *PublicEthereumAPI) MaxPriorityFeePerGas(ctx context.Context) (*hexutil.
 	return (*hexutil.Big)(tipcap), err
 }
 
-type feeHistoryResults struct {
-	FirstBlock   rpc.BlockNumber
-	Reward       [][]*hexutil.Big
-	BaseFee      []*hexutil.Big
-	GasUsedRatio []float64
+type feeHistoryResult struct {
```
Comment (zsfelfoldi) [inline internal/ethapi/api.go:83]: Also my mistake, but the field names have been changed in the spec last minute which I forgot to update in the code:
"firstBlock" is now called "oldestBlock" and "baseFee" is now called "baseFeePerGas".
https://github.com/ethereum/eth1.0-specs/blob/master/json-rpc/spec.json#L246

**Review by zsfelfoldi — APPROVED** [review]
LGTM, see comment about result field names

File: internal/ethapi/api.go (hunk 2 above)
Comment (karalabe) [reply internal/ethapi/api.go:83]: Fixed, PTAL

File: eth/gasprice/feehistory.go (hunk 1 above)
Comment (karalabe) [reply eth/gasprice/feehistory.go]: Fixed, PTAL
